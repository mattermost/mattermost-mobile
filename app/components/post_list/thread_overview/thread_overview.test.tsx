// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {renderWithIntl} from '@test/intl-test-helper';

import ThreadOverview from './thread_overview';

import type PostModel from '@typings/database/models/servers/post';

describe('ThreadOverview', () => {
    it('should match snapshot when post is saved with no replies', () => {
        const props = {
            isSaved: true,
            repliesCount: 0,
            rootId: '',
            rootPost: {} as PostModel,
            testID: 'thread-overview',
        };

        const wrapper = renderWithIntl(<ThreadOverview {...props}/>);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should match snapshot when post is saved and has replies', () => {
        const props = {
            isSaved: false,
            repliesCount: 2,
            rootId: '',
            rootPost: {} as PostModel,
            testID: 'thread-overview',
        };

        const wrapper = renderWithIntl(<ThreadOverview {...props}/>);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
