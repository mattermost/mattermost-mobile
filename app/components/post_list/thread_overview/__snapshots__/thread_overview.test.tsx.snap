// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadOverview should match snapshot when post is saved and has replies 1`] = `
<View
  style={
    [
      {
        "borderBottomWidth": 1,
        "borderColor": "rgba(63,67,80,0.1)",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "marginVertical": 12,
        "paddingHorizontal": 16,
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
  testID="thread-overview"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "rgba(63,67,80,0.64)",
          "fontFamily": "OpenSans",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
          "marginHorizontal": 4,
        }
      }
      testID="thread-overview.replies_count"
    >
      2 replies
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      delayLongPress={600}
      enabled={true}
      exclusive={true}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      innerRef={null}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        [
          undefined,
          {
            "cursor": undefined,
          },
        ]
      }
      testID="thread-overview.save.button"
      touchSoundDisabled={false}
    >
      <View
        accessible={true}
        collapsable={false}
        style={
          {
            "marginLeft": 16,
            "opacity": 1,
          }
        }
      >
        <Icon
          color="rgba(63,67,80,0.64)"
          name="bookmark-outline"
          size={24}
        />
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      delayLongPress={600}
      enabled={true}
      exclusive={true}
      handlerTag={2}
      handlerType="NativeViewGestureHandler"
      innerRef={null}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        [
          undefined,
          {
            "cursor": undefined,
          },
        ]
      }
      testID="thread-overview.post_options.button"
      touchSoundDisabled={false}
    >
      <View
        accessible={true}
        collapsable={false}
        style={
          {
            "marginLeft": 16,
            "opacity": 1,
          }
        }
      >
        <Icon
          color="rgba(63,67,80,0.64)"
          name="dots-horizontal"
          size={24}
        />
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;

exports[`ThreadOverview should match snapshot when post is saved with no replies 1`] = `null`;
